proposed/container.md

```diff
@@ -21,6 +21,12 @@ Users of dependency injections containers (DIC) are referred to as `user`.
 
 ### 1.1 Basics
 
+#### 1.1.1 Entry Identifiers
+
+An entry identifier is a string of at least one character that uniquely identifies an item within a container.  Implementing libraries MUST support identifiers consisting of any valid non-empty string of arbitrary length in UTF-8 encoding.  Implementing libraries MAY support additional encodings, but must support at least UTF-8 at minimum.  An entry identifier is an opaque string, so callers SHOULD NOT assume that the structure of the string caries any semantic meaning.
```

Comment:
@Jean85 could you elaborate what's the meaning of "opaque string" in this case? Either I'm not really good with English or it is quite vague...